lib/opencensus/trace/integrations/faraday_middleware.rb

```diff
@@ -63,12 +63,28 @@ module Integrations
       #
       # ## Trace context
       #
-      # This currently adds a "Trace-Context" header to each outgoing request,
-      # propagating the trace context for distributed tracing. See
-      # https://github.com/TraceContext/tracecontext-spec for more information
-      # on the trace context.
+      # This currently adds a header to each outgoing request, propagating the
+      # trace context for distributed tracing. By default, this uses the
+      # formatter in the current config.
       #
-      # TODO: Support other trace context formats.
+      # You may provide your own implementation of the formatter by configuring
+      # it in the middleware options hash. For example:
+      #
+      #     conn = Faraday.new(url: "http://www.example.com") do |c|
+      #       c.use OpenCensus::Trace::Integrations::FaradayMiddleware,
+      #             formatter: OpenCensus::Trace::Formatters::CloudTrace.new
```

Comment:
At this point, I wonder if the term "propagator" is more clear on the role of this object.

Reply:
Unless it's actually injecting the context, I don't think it's a "propagator". If we make it inject context into the outgoing request, or extract it from the incoming request, then maybe it's a propagator. As it is now, the rack middleware and faraday middleware are more of the propagators than anything else.